src/main/java/org/opensearch/knn/index/codec/util/CodecBuilder.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.knn.index.codec.util;
+
+import lombok.AllArgsConstructor;
+import org.apache.lucene.codecs.Codec;
+import org.opensearch.index.mapper.MapperService;
+import org.opensearch.knn.index.codec.KNN910Codec.KNN910Codec;
+
+/**
+ * Abstracts builder logic for plugin codecs
+ */
+public abstract class CodecBuilder {
```

Comment:
what is the exact purpose this interface can't we just create a utility function to achieve what we are doing here in this interface and its implementation.

Given the logic to init the codec is pretty straight forward.

Reply:
I think utility function will create more tight coupling, and I like to have more clear abstraction in this area. We going to have more codecs in future, typically it's for each major core and lucene release, plus our interim releases in case we have some development in codec classes. Version 9.1.0 is the first one where we added codec factory so it look a bit overkill for one codec now, but it will pay off in future when we have more codec versions.

Reply:
I think we even need to add new codec based on Lucene92 for lucene hnsw to have foundation for pre-filter support